File: src/main/java/rst/wordpress/Page.java
package rst.wordpress;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

/**
 * Created by Administrator on 2016-09-23.
 */
public class Page  {
    public final WebDriver driver;
    public String baseUrl = "https://automatyzacja2016.wordpress.com/";
    public Page(WebDriver driver) {
        this.driver = driver;
    }

    protected void insertText(By where, String text) {
        //new WebDriverWait(driver,5).until(ExpectedConditions.visibilityOfAllElementsLocatedBy(where));
        driver.findElement(where).clear();
        driver.findElement(where).sendKeys(text);
    }
    protected void clickElement(By where) {
        new WebDriverWait(driver,5).until(ExpectedConditions.elementToBeClickable(where));
        driver.findElement(where).click();
    }

    public void closeBrowser() {
        driver.get(baseUrl + "/wp-admin");
        clickElement(By.cssSelector("img.avatar.avatar-32"));
        //new WebDriverWait(driver,2).until(ExpectedConditions.elementToBeClickable(By.xpath("//*[@id=\"wp-admin-bar-user-info\"]/div/form/button")));
        //KliknijXpath("//*[@id=\"wp-admin-bar-user-info\"]/div/form/button");

        new WebDriverWait(driver,2).until(ExpectedConditions.elementToBeClickable(By.cssSelector("button.ab-sign-out")));
        clickElement(By.cssSelector("button.ab-sign-out"));
    }
}

File: TestAutomation/Test/Page.cs
using System;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading;
using NUnit.Framework;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Chrome;

namespace TestAutomation.Test
{
    internal class Page
    {
        protected IWebDriver driver;
        protected string baseURL = "https://automatyzacja2016.wordpress.com/";
        public Page(IWebDriver driver)
        { 
        this.driver = driver;
        }

        protected void Click(By by)
        {
            
            WebDriverWait Wait = new WebDriverWait(driver, new TimeSpan(0, 0, 10));
            Wait.Until(ExpectedConditions.ElementToBeClickable(by));
            driver.FindElement(by).Click();
        }

        protected void InsertText(By by, string text)
        {
            
            driver.FindElement(by).Clear();
            driver.FindElement(by).SendKeys(text);
        }
    }
}
